### ISSUE TITLE
Pin collector image in e2e tests

### ISSUE THREAD

**Issue by MrAlias** [open-telemetry/opentelemetry-go-instrumentation#1441]
We currently use the latest collector image for our testing:

https://github.com/open-telemetry/opentelemetry-go-instrumentation/blob/d447af1ffa70611a8393f9c189f927e54fe479b4/.github/workflows/e2e/k8s/collector-helm-values.yml#L31

This means any breaking changes to the image will be introduced to our CI system without review (e.g. https://github.com/open-telemetry/opentelemetry-go-instrumentation/actions/runs/12377825359/job/34549117158?pr=1431)

We need to pin this image. We also need to ensure that renovate can help in the upgrade process.

**Comment by MrAlias** [comment]
Related to https://github.com/open-telemetry/opentelemetry-collector-releases/issues/783

**Comment by MrAlias** [comment]
#1443 provides a temporary fix. We still need to develop a solution that will allow renovate to update the collector image version.

Likely, this means wrapping the collector image in our own Dockerfile.

**Comment by MrAlias** [comment]
> #1443 provides a temporary fix. We still need to develop a solution that will allow renovate to update the collector image version.
> 
> Likely, this means wrapping the collector image in our own Dockerfile.

Possible solution would be to follow what is planned here: https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1140/files#diff-292d6d00bcf2d0d55e146ba46e229366f0ed73fcb0ac11c4885a029f014161bb

**Comment by txabman42** [comment]
> Possible solution would be to follow what is planned here: https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/1140/files#diff-292d6d00bcf2d0d55e146ba46e229366f0ed73fcb0ac11c4885a029f014161bb

Just an idea for this:
Renovabot's custom managers with regex would allow to continue depending on official otel contrib images for e2e tests

https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/2072

**Comment by txabman42** [comment]
Could be closed this issue? Renovabot successfully upgraded contrib version in e2e tests (https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/2086)

**Comment by MrAlias** [comment]
Yes. This was resolved and then it became irrelevant with https://github.com/open-telemetry/opentelemetry-go-instrumentation/pull/2118.
